File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs)
{
    size_t length = strlen(lhs);

    if (strlen(rhs) != length)
        return INVALID;

    int count = 0;

    for (int i = 0; i < (int)length; i++)
    {
        if (lhs[i] != rhs[i])
            count++;
    }

    return count;
}

File: solutions/javascript/hamming/1/hamming.js
export const compute = (a, b) => {
  if (a.length !== b.length) {
    throw new Error('strands must be of equal length');
  }

  let count = 0;

  for (let i = 0; i < a.length; i++) {
    if (a[i] !== b[i]) {
      count++;
    }
  }

  return count;
};
